docs/gallery/teapots.ipynb

```diff
@@ -0,0 +1,105 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1f73b51c-057d-4969-8c3d-6cbe009bd93f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import plopp as pp\n",
+    "import scipp as sc\n",
+    "import numpy as np\n",
+    "from plopp.data.mesh import teapot"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "17405423-0496-40d5-973c-c7ec6aa42746",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def make_one_teapot_mesh3d(**kwargs):\n",
+    "    mesh, faces, names = teapot()\n",
+    "    return pp.mesh3d(mesh, faces, **names, **kwargs)"
```

Comment:
So this is different from the interface I had in mind.
I know it is modeled on the interface to `scatter3d`, but I am starting to wonder if we should also change that.

I thought one could maybe have something that would resemble more what the mesh needs, like
```Py
pp.mesh3d(vertices=sc.array(shape=N, dtype=vector3),
          faces=sc.array(shape=[N, 3]),
          color=sc.array(shape=N, dtype=float))
```
Internally, a bit like [xyplot](https://github.com/scipp/plopp/blob/main/src/plopp/plotting/xyplot.py), the `vertices, faces, color` inputs could be wrapped into nodes, before supplying them to the figure, so that if one of them updates, it would notify the figure.

Only downside is that this makes it more difficult to plot multiple meshes (on scatter3d, you can supply a dict of data arrays to plot two point clouds on the same scene).

Not sure what a good API would be for that, but I would like to use a similar API for the normal scatter plot:
```Py
pp.scatter3d(x=array_x, y=array_x, z=array_x, color=array_data)
```

Reply:
Is there a reason why we the mesh needs to be made of N identical objects? Sure that's what it would be like for detector pixels of an instrument, but really if you are making something that can plot a mesh, you should be free from that restriction?

I was seeing it as something just very generic, that would show a mesh. The fact that we construct it with `NxV` vertices would be done externally (e.g. in `essspectroscopy`).
Then the colors for the vertices would also need to be constructed there (also `NxV` in size).

That said, this would then make it much more difficult to get it to work with the plane slicing.
I guess for the bifrost instrument view, having such slicing would be useful?
One could say that if you really need the slicing, you have to stick to the simple point cloud.

Reply:
My initial thought seeing this was that `N=1` is a special case already handled by the implementation that could be extended to an _implicit_ `N=1` if that dimension is missing from the input.
But I think I now understand your point, and may even understand the motivation for allowing the three vector components of the vertices to be specified separately.

---

My concern is that plotting intensity on `N` topologically-equivalent polyhedra is not a BIFROST, spectroscopy, or cylindrical-detector-instrument specific task; and that converting from `N` &times; `V` vertices and `N` intensities to their flattened lists is not _entirely trivial_, especially for someone unfamiliar with the dimension traversal order of `scipp.flatten`.

Would the `plopp`-way of achieving this be an extra node to handle the conversion before your desired-signature `mesh3d`?
Can that node be part of `plopp` to avoid duplication and the chance for errors that brings?